### ISSUE TITLE
Add citar-add-library-file

### ISSUE THREAD

**Issue by bdarcus** [bdarcus/citar#528]
I added a modified, more general, version of this to the wiki. I may add it back to citar; I just wasn't sure if anyone used this ;-)

https://github.com/bdarcus/citar/wiki/Example-functions#add-file-to-library

There's still a little bug in it that I'll fix.

It uses `completing-read` rather than `completing-read-multiple`, and works on any file.

Note: recent changes in many of these commands are designed to make them compatible with `embark-act-all`. I'm not sure the UI on the current version of this command actually is, seems it seems to prompt you to rename the file in one case. Worth testing this.

It also occurs to me that we might want to include an option to add from a DOI, which maybe does that without prompting if the entry has a DOI?

_Originally posted by @bdarcus in https://github.com/bdarcus/citar/discussions/522#discussioncomment-1924933_

**Comment by suyashbire1** [comment]
Hi, thank you for this very useful package! 

I recently switched to vertico from ivy. I used this feature very frequently with ivy-bibtex so it would be useful to have it in citar as well. Thank you very much!

Also, note that `file-name-concat` function requires emacs version 28.1. It wasn't working for me in 27.2. Just putting it out there if someone else is facing the same issue.

> It also occurs to me that we might want to include an option to add from a DOI, which maybe does that without prompting if the entry has a DOI?

You mean add a pdf? That would be cool!
If you mean add bibtex entry from doi, `biblio` already has a similar function.
